core/src/main/java/org/apache/accumulo/core/client/ScannerBase.java

```diff
@@ -390,4 +390,21 @@ default Stream<Entry<Key,Value>> stream() {
     return StreamSupport.stream(this.spliterator(), false);
   }
 
+  /**
+   * Set user data on the Scanner. This data will be added to server side information and logs for
+   * correlation.
+   *
+   * @param userData meaningful data that can be used to correlate server side information
+   * @since 3.0.0
+   */
+  void setUserData(String userData);
```

Comment:
"tag" may not be the best choice - in the terms of metrics, tag has a connotation of creating a time-series. Not sure if using tag in this context would lead to that conclusion, but wanted to point it out for discussion.

Reply:
Metrics wasn't my first thought at all. The term is generically used to refer to any kind of auxiliary descriptive mechanism that enables automated processing. So, it's used in a lot of similar contexts. Even though the term is a generic one, I was thinking it might be more intuitive than the less familiar "user data" which, unlike "tag", doesn't intuitively lead one to think about auxiliary descriptive information for collating. (NIST definition is https://csrc.nist.gov/glossary/term/data_tag)

Reply:
Initially I was thinking about adding this to the MDC, but not all frameworks work with that.

Reply:
It looks like `correlationID` is a common term, with tracing systems using the term `traceId` as their correlationID. See: https://microsoft.github.io/code-with-engineering-playbook/observability/correlation-id/. Could rename to `scanCorrelationId`.

Reply:
I like `setCorrelationId(String correlationId)`.  Maybe `scan` is not needed in the name since its being set on a scanner.

Reply:
I renamed to correlationId in 84b9cbe